### PR TITLE
Add Button to Dynamically Create Polaroids

### PR DESCRIPTION
**Pull request by amuamushu**
https://screenshot.googleplex.com/9Wzu9D2zNxX.png

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+  // Creates a div tag to store the polaroid image and description.
+  let div = document.createElement("div");
+  div.setAttribute('class', 'polaroid');
+
+  // Creates a p tag that contains the image name.
+  const imageName = image.split(".")[0];
+  let imageText = createPTag(imageName);
+
+  // Creates an img tag to store the image.
+  let imgTag = createImgTag(image);
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:39]: Should this be a const?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+
+  div.appendChild(imgTag);
+  div.appendChild(imageText);
+  
+  // Adds the polaroid div tag to the page.
+  const imagesContainer = document.getElementById('images');
+  imagesContainer.appendChild(div);
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:46]: How can this be const if you append div?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+  // Picks a random image.
+  const image = images[Math.floor(Math.random() * images.length)];
+  
+  // Creates a div tag to store the polaroid image and description.
+  let div = document.createElement("div");
+  div.setAttribute('class', 'polaroid');
+
+  // Creates a p tag that contains the image name.
+  const imageName = image.split(".")[0];
+  let imageText = createPTag(imageName);
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:36]: can this be const?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+  div.appendChild(imgTag);
+  div.appendChild(imageText);
+  
+  // Adds the polaroid div tag to the page.
+  const imagesContainer = document.getElementById('images');
+  imagesContainer.appendChild(div);
+}
+
+/**
+ * Creates an img tag to store the image.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:50]: nit: where is it storing the image?

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+ * @param {string} image The name of the image
+ *     file to create an img tag for.
+ * @return {object} This returns an img tag using
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:53]:
```suggestion
 * @return {object} Returns an img tag using
```

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+ *     the given image.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:54]:
```suggestion
 *     the given image file name.
```

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+ */
+function createImgTag(image) {
+  let imgTag = document.createElement("img");
+  imgTag.setAttribute("src", 'images/' + image);
+  
+  const imageName = image.split(".")[0];
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:60]: Can you create a global const for "." given it is used in many places?

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+/**
+ * Creates an img tag to store the image.
+ * @param {string} image The name of the image
+ *     file to create an img tag for.
+ * @return {object} This returns an img tag using
+ *     the given image.
+ */
+function createImgTag(image) {
+  let imgTag = document.createElement("img");
+  imgTag.setAttribute("src", 'images/' + image);
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:58]: the path "images" seems to be used here and on line 45. It would be good to store this in a global const because if you ever change the directory you do not need to change it in multiple places.

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+  const imageName = image.split(".")[0];
+  imgTag.setAttribute("alt", imageName);
+  return imgTag;
+}
+
+/**
+ * Creates an p tag to store the given text.
+ * @param {string} text The text to include in
+ *     the p tag.
+ * @return {object} This returns an p tag using
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:69]:
```suggestion
 * @return {object} Returns an p tag using
```

File: portfolio/src/main/webapp/script.js [hunk 10]
```diff
@@ -16,28 +16,70 @@ const VISIBLE = 'visible';
...
+ *     the given image.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:70]:
```suggestion
 *     the given text.
```

File: portfolio/src/main/webapp/script.js (hunk 9 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: done!

File: portfolio/src/main/webapp/script.js (hunk 10 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: done!

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js:47]: I believe declaring a variable as a constant just prevents reassignment. The object would still be mutable, which is how I was able to append to imagesContainer without a type error.

I updated it to be const anyways for safety.
Done

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js:51]: I intended for the location to be ambiguous because I assigned the location in the method that calls createImgTag. Is that okay?

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: done!

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: It doesn't get updated later so I can make it a const
done!

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: Done!

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: Done!

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (billie-cat) [reply portfolio/src/main/webapp/script.js:51]: Yes that is ok. Maybe rephrase it: Creates an img tag needed for storing the image.

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: done!

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: The 'images' in this line refer to a folder while the images in line 45 refer to a HTML element. Since they refer to different things, should I still create a global const for them?

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (billie-cat) [reply portfolio/src/main/webapp/script.js:47]: Oh if you can keep it const, then do that. Also do div, that seems it can be const?

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -14,30 +14,73 @@
...
+  let imgTag = document.createElement('img');
+  imgTag.setAttribute('src', 'images/' + image);
+  
+  const imageName = image.split(SEPARATOR)[0];
+  imgTag.setAttribute('alt', imageName);
+  return imgTag;
+}
+
+/**
+ * Creates an p tag to store the given text.
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:67]:
```suggestion
 * Creates a p tag to store the given text.
```

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js:47]: done!

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: done!

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: (I updated this is in pull request #19)

Done!
